src/Servers/Kestrel/shared/KnownHeaders.cs

```diff
@@ -288,7 +290,7 @@ static string AppendHPackSwitchSection(HPackGroup group)
             }
         }
 
-        static string AppendSwitchSection(int length, IOrderedEnumerable<KnownHeader> values)
+        static string AppendSwitchSection(int length, IList<KnownHeader> values)
```

Comment:
Shouldn't we just remove the HTTP/2-only headers from AppendSwitchSection entirely since it's handled by TryHPackAppend?

Reply:
It is still needed. A client doesn't have to use HPack. They could send header with ":authority" as name bytes.